test/TestResourcesAndFunctions.cpp

```diff
@@ -0,0 +1,259 @@
+/**
+ * Copyright (c) 2018 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#include "TestResourcesAndFunctions.h"
+#include <cstdio>
+#include <cstring>
+#include <experimental/filesystem>
+#include <iostream>
+#include <UnitTest++/UnitTest++.h>
+#include <common/IOHelper.h>
+
+using namespace std::experimental::filesystem;
+using std::experimental::filesystem::path;
+
+mutex TestResourcesAndFunctions::staticLock;
+
+vector<string> TestResourcesAndFunctions::testVectorWithSimulatedDecompressedBlockData;
+
+TestResourcesAndFunctions::TestResourcesAndFunctions(string testSuite, string testName) {
+    this->testSuite = move(testSuite);
+    this->testName = move(testName);
+    if (this->testSuite.empty())
+        this->testSuite = "default";
+}
+
+TestResourcesAndFunctions::TestResourcesAndFunctions(string testName) {
+    this->testSuite = "default";
+    this->testName = move(testName);
+}
+
+TestResourcesAndFunctions::~TestResourcesAndFunctions() {
+    finalize();
+}
+
+void TestResourcesAndFunctions::finalize() {
+    lock.lock();
+    if (!testPath.empty())
+        remove_all(testPath);
+    testPath = "";
+    lock.unlock();
+}
+
+path TestResourcesAndFunctions::getTestPath() {
+    lock.lock();
+    if (testPath.empty()) {
+        string prefix = "FastqIndExTest_" + testSuite + "_" + testName;
+        auto[result, tempDir] = IOHelper::createTempDir(prefix);
+        if (!result) {
+            cerr << "## The temporary folder for pattern '" << prefix << "' could not be created!\n";
+        }
+        bool success = exists(tempDir);
+                CHECK(!tempDir.string().empty());
+                CHECK(success);
+        testPathCreationWasSuccessful = success;
+        testPath = tempDir;
+    }
+    lock.unlock();
+    return testPath;
+}
+
+string TestResourcesAndFunctions::getTestSuite() {
+    return testSuite;
+}
+
+string TestResourcesAndFunctions::getTestName() {
+    return testName;
+}
+
+bool TestResourcesAndFunctions::getTestPathCreationWasSuccessful() {
+    return testPathCreationWasSuccessful;
+}
+
+path TestResourcesAndFunctions::filePath(const string &filename) {
+    return getTestPath().string() + "/" + filename;
+}
+
+path TestResourcesAndFunctions::getResource(const string &filename) {
+    // Utilizing current_path() will result in the path of current test-binary, which will be e.g.:
+    //   ~/Projects/FastqIndEx/cmake-build-debug/test/testapp
+    // As we do want resources to be loaded from:
+    //   ~/Projects/FastqIndEx/cmake-build-debug/test/testapp
```

Comment:
This line does not differ from the previous path line. What do you mean?
```suggestion
    //   ~/Projects/FastqIndEx/test/resources/
```
???